packages/server-wallet/src/chain-service/chain-service.ts

```diff
@@ -108,13 +117,13 @@ export class ChainService implements ChainServiceInterface {
       ContractArtifacts.NitroAdjudicatorArtifact.abi
     );
 
-    this.nitroAdjudicator.on(ChallengeRegistered, (...args) => {
+    this.subscribeToConfirmedEvents(this.nitroAdjudicator, ChallengeRegistered, (...args) => {
```

Comment:
In this case, the chain service should probably notify subscribers as soon as the chain service observes the challenge event. Challenges are time sensitive. We should give subscribers the maximum amount of time to process a challenge.

Reply:
will do
nitroAdjudicator should not use confirmed events then

Reply:
I think the intuition to use unconfirmed events for nitro adjudicator is probably correct. Chain service waits for a confirmation of a funding event to protect against the following scenario:
1. Users A and B hold prefund states 0 and 1.
2. User A deposits on block 0.
3. User B deposits on block 1.
4. User A signs postfund 0.
5. User B signs postfund1 and starts the state channel application.
6. User B pays 1 eth to user A in the state channel.
5. There is a chain reorg, and block 1 is no longer a part of the chain. Note that now User B has not deposited their share of the funds. So the payment above cannot be redeemed.